presto-spi/src/main/java/com/facebook/presto/spi/session/SessionConfigurationContext.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.session;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.util.Collections.unmodifiableSet;
+import static java.util.Objects.requireNonNull;
+
+public final class SessionConfigurationContext
+{
+    private final String user;
+    private final Optional<String> source;
+    private final Optional<String> catalog;
+    private final Optional<String> schema;
+    private final Set<String> clientTags;
+
+    public SessionConfigurationContext(String user, Optional<String> source, Optional<String> catalog, Optional<String> schema, Set<String> clientTags)
+    {
+        this.user = requireNonNull(user, "user is null");
+        this.source = requireNonNull(source, "source is null");
+        this.catalog = requireNonNull(catalog, "catalog is null");
+        this.schema = requireNonNull(schema, "schema is null");
+        this.clientTags = unmodifiableSet(new HashSet<>(requireNonNull(clientTags, "clientTags is null")));
+    }
+
+    public String getUser()
+    {
+        return user;
+    }
+
+    public Optional<String> getSource()
+    {
+        return source;
+    }
+
+    public Optional<String> getCatalog()
```

Comment:
Catalog and schema should not be here. These are only used for resolving unqualified names in SQL statements. According to the SQL standard, this should actually be a search path, which we will likely implement at some point.

Reply:
If that's the case, we need to implement that now then. One concrete use case for this feature we have is to be able to apply different session properties to interactive vs. batch clusters. Without catalog that would be hard to do. I would suggest keeping this for now and refactor this when search path is actually implemented.

Reply:
We know which cluster is which when the cluster is configured, so this can be part of the configuration for the plugin.

Reply:
Consider the following query:
```sql
SELECT * FROM foo.x.y JOIN bar.x.y
```
The query accesses two different catalogs, `foo` and `bar`.

Reply:
What's the point of having catalog and schema in session then? If that would not necessarily have anything to do with the query itself? It the use case is to route the query to a certain cluster, then here it's the same. If this query would be routed to the cluster serving `zzz`, then we want to apply default session settings for the cluster serving `zzz`, which has nothing to do with `foo` or `bar`.

Reply:
They are needed in the session because that's how it gets to the analyzer, which uses it to resolve names. It's not in `ConnectorSession` because it isn't relevant after analysis.